src/php/Lang/PhelArray.php

```diff
@@ -138,7 +138,18 @@ public function hash(): string
 
     public function equals($other): bool
```

Comment:
Not really sure about the last part of the method. In my opinion this would be enough:

```php
if (!($other instanceof PhelArray)) {
  return false;
}

return $this->data === $other->data;
```

Have you an example where this fails?

Reply:
I tried that first, sure, and here is what I got :)
<img width="1121" alt="Screenshot 2021-01-29 at 09 05 07" src="https://user-images.githubusercontent.com/5256287/106248304-17e85400-6211-11eb-914f-b872143cd611.png">

Reply:
Ufff. Then I should have a look at that. Maybe we just need to call `reset` on both arrays before:

```php
if (!($other instanceof PhelArray)) {
  return false;
}

reset($this->data);
reset($other->data);

return $this->data === $other->data;
```

Reply:
Clever try, but still:

<img width="1119" alt="Screenshot 2021-01-29 at 09 19 05" src="https://user-images.githubusercontent.com/5256287/106249752-0bfd9180-6213-11eb-845f-bec6a9dbdbe1.png">

Reply:
Ok, I will have a look